verilog/formatting/formatter.cc

```diff
@@ -266,12 +281,47 @@ verible::util::Status Formatter::Format(const ExecutionControl& control) {
   const auto unwrapped_lines =
       MakeUnwrappedLinesWorklist(*format_tokens_partitions, style_);
 
+  std::vector<UnwrappedLine> fitted_lines;
+  fitted_lines.reserve(unwrapped_lines.size());
+  int fitted_lines_indent = -1;
+  for (const auto& uwline : unwrapped_lines) {
```

Comment:
//explain this block of code
Looks like you're implementing the `kAppendFittingSubpartitions` policy, and keeping track of the wrapping column position, and greedily appending one subpartition at a time.

Reply:
I've rewritten whole code and added more comments. I hope this version is easier to read/understand.